### DISCUSSION TITLE
Get sites and folders

### DISCUSSION THREAD

**Discussion by mar4elom** [pnp/pnpcore#538] [Q&A]
Hi,

I have no idea how to get sites and folders in our company Sharepoint account.
I use Demo.WPF example and PnP Core SDK documentation to get sites and folders, but get one weird result.
If I try to get site by name I get error "SharePoint Rest service exception".
If I log in through browser I can see many sites and a lot of folders inside.
Please help!

**Comment by jansenbe** [comment]
@mar4elom : the library currently does not have support for numerating site collections as that's what I assume you need. That functionality is however coming fairly soon (early October).

If you know the urls of the site collections your needs to run on you can just use that url to create a context. See https://pnp.github.io/pnpcore/using-the-sdk/readme.html#using-the-pnpcontext-for-operations-on-microsoft-365 for some samples or below:

```csharp
using (var context = await pnpContextFactory.CreateAsync(new Uri("https://contoso.sharepoint.com/sites/asite"))
{
    var web = await context.Web.GetAsync();
    Console.WriteLine($"Title: {web.Title}");
}
```

It also helps if you share the code you're writing and the error you get :-)

**Reply by mar4elom** [reply]
It looks like I need library functionality, so I should wait till October.
Thank you for your support!

**Reply by jansenbe** [reply]
@mar4elom : support for enumerating all site collections is there in preview: see https://pnp.github.io/pnpcore/using-the-sdk/admin-sharepoint-sites.html#getting-all-site-collections
